From 97a5f6039458a57e6c05bb910c104ac2ecee7622 Mon Sep 17 00:00:00 2001
From: jsg <jsg@openbsd.org>
Date: Wed, 27 Oct 2021 02:27:34 +0000
Subject: [PATCH] backport a perl patch to avoid excessive warnings with llvm
 13 clang

originally from Tony Cook in
skip using gcc brace groups for STMT_START/END
7169efc77525df70484a824bff4ceebd1fafc760

looks fine millert@ ok afresh1@
---
 # The PPPort patch is no longer needed, we upgraded it
 gnu/usr.bin/perl/dist/Devel-PPPort/parts/inc/misc | 9 ++-------
 gnu/usr.bin/perl/perl.h                           | 5 -----
 2 files changed, 2 insertions(+), 12 deletions(-)

diff --git perl.h perl.h
index b0d35e27c724..76f3ba151c65 100644
--- perl.h
+++ perl.h
@@ -662,13 +662,8 @@ out of them.
  Trying to select a version that gives no warnings...
 */
 #if !(defined(STMT_START) && defined(STMT_END))
-# ifdef PERL_USE_GCC_BRACE_GROUPS
-#   define STMT_START	(void)(	/* gcc supports "({ STATEMENTS; })" */
-#   define STMT_END	)
-# else
 #   define STMT_START	do
 #   define STMT_END	while (0)
-# endif
 #endif
 
 #ifndef BYTEORDER  /* Should never happen -- byteorder is in config.h */
